modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountEntryLocalServiceImpl.java

```diff
@@ -528,6 +528,18 @@ private void _populateSearchContext(
 			params.get("status"), WorkflowConstants.STATUS_APPROVED);
 
 		searchContext.setAttribute("status", status);
+
+		String type = (String)params.get(Field.TYPE);
+
+		if (Validator.isNotNull(type)) {
+			String[] types = {type};
+
+			if (type.equals("all")) {
+				types = AccountConstants.ACCOUNT_ENTRY_TYPES;
+			}
+
+			searchContext.setAttribute(Field.TYPE, types);
+		}
```

Comment:
This works too, but I'd prefer not to set the type attribute for `all` as `all` means any type, so no filter is needed. I'd do this to simplify the search query.

```
if (Validator.isNotNull(type) && !type.equals("all")) {
    searchContext.setAttribute(Field.TYPE, type);
}
```